src/python/gudhi/off_utils.pyx

```diff
@@ -25,21 +26,95 @@ __license__ = "MIT"
 cdef extern from "Off_reader_interface.h" namespace "Gudhi":
     vector[vector[double]] read_points_from_OFF_file(string off_file)
 
+def _get_next_line(file_desc, comment='#'):
+    """Return the next line that is not a comment.
+
+    :param file_desc: An open file in read mode.
+    :type file_desc: A file descriptor
+    :param comment: The characters or list of characters used to indicate the start of a comment.
+    :type comment: string
+
+    :returns: The next line.
+    :rtype: string
+    """
+    while True:
+        # file_desc.readline() is preferred to next(file_desc), as the second option is not compatible with
+        # seek and tell methods
+        line = file_desc.readline()
+        if not line.startswith(comment):
+            break
+    return line
+
+def _read_off_file_header(input_file):
+    """Return the information contained in the header of an OFF file.
+
+    :param off_file: An OFF file style name.
+    :type off_file: string
+
+    :returns: The point cloud dimension, and the number of points.
+    :rtype: tuple(int, int)
+
+    Raises:
+        ValueError: If the file does not respect the OFF file format.
+    """
+    nb_vertices = -1
+
+    line = _get_next_line(input_file)
+    # First line should be "OFF" (3d case) or "nOFF" (dD case)
+    if line.lower().startswith("off"):
+        dim = 3
+    elif line.lower().startswith("4off"):
+        dim = 4
+    elif line.lower().startswith("noff"):
+        # "nOFF" case, next line is the dimension
+        # can also contain nb_vertices, nb_faces nb_edges (can also be on the next line)
+        line = _get_next_line(input_file)
+        digits = [int(s) for s in line.split() if s.isdigit()]
+        dim = digits[0]
+        if len(digits) > 1:
+            nb_vertices = digits[1]
+            # nb_faces =  digits[2]
+            # nb_edges =  digits[3] # not used - can be ignored
+            # nb_cells =  digits[4]
+    else:
+        raise ValueError(f"Inconsistent OFF header, got '{line.rstrip()}', should be 'OFF', '4OFF' or 'nOFF'")
+        
+    # nb_vertices can be already set by "nOFF" case, when 'dim nb_vertices nb_faces nb_edges' on the same line
+    if nb_vertices < 0:
+        # Number of points is the first number ("OFF" case) or the second one ("nOFF" case) of the second line
+        line = _get_next_line(input_file)
+        digits = [int(s) for s in line.split() if s.isdigit()]
+        nb_vertices = digits[0]
+        # nb_faces =  digits[1]
+        # nb_edges =  digits[2] # not used - can be ignored
+        # nb_cells =  digits[3]
+    # If comments between header and the first point
+    line = _get_next_line(input_file)
```

Comment:
Doesn't loadtxt already ignore comments?

Reply:
Ah, so the new code fails to handle comments when numpy is too old, but you added a hack so it still does ok as long as the comments are in the most likely places? That makes sense. Could you add a comment next to this line? (we'll want to remove this at some point)
> It is advised to use a numpy version ≥ 1.23.0 if the `off_file` contains lines without data or with comments in between the points.

"advised" may not be strong enough. Empty/comment lines between the points are only supported with numpy ≥ 1.23.0.

Reply:
Modify the comment and the documentation warning on bd0f3feff